### ISSUE TITLE
What's the problem with de, zh and ja in the heatmap?

### ISSUE THREAD

**Issue by asmaier** [Babylonpartners/fastText_multilingual#3]
Why do the german (de), chinese (zh) and japanese (ja) language standout from the other languages in the heatmap ? The inter-pair precision of these languages into any other language seems to be very low. Is this a problem with the analysis or is this a real effect?

**Comment by sls56** [comment]
Hi asmaier, I think this is at least partly a technical issue in the analysis. We're currently away on conference/holiday, but I will look into it tuesday/wednesday and get back to you :)

**Comment by tzatter** [comment]
Japanese language is different from English. You need to separate words with [mecab](http://taku910.github.io/mecab/). Japanese words doesn't have spaces among words

**Comment by nhammerla** [comment]
@tzatter  We will investigate next week and update the repo accordingly. We used a simple regexp (`r'\b\w+\b'`) to "tokenize" translations that weren't in the target dictionary. I'm not sure how clever the `re` implementation is when dealing with non-latin texts so some of the error might be from that!

**Comment by tzatter** [comment]
@nhammerla hello, I am Japanese. Japanese language is not such easy haha. I know japanese model of gensim instead of fasttext. Please refer to the source code and pre-trained model
source code is https://github.com/shiroyagicorp/japanese-word2vec-model-builder
pre-trained model is http://public.shiroyagi.s3.amazonaws.com/latest-ja-word2vec-gensim-model.zip

good luck!

**Comment by sls56** [comment]
Hi asmaier and tzatter,
So I investigated the Chinese and Japanese training/test set word vectors, and they have very high self similarity meaning that, for example, 2 randomly chosen Japanese words from our dictionary have a mean cosine similarity of 0.85. This explains why they stand out on the performance matrix, and suggests there is probably a bug in our code to extract word vectors from the dictionaries of these languages; which as Nils suggested is likely caused by issues when tokenising non-arabic text.

We are unfortunately quite busy with commercial work for the babylon app at the minute but we will try to fix this as soon as we can! We also want to release training/test code, and to check whether the Google Translate API allows us to open-source the dictionaries themselves.

This problem doesn't apply to German, whose self similarity was normal (around 0.16), and I haven't been able to find any other issues here yet.
